tools/text_processing/text_processing/awk.xml

```diff
@@ -1,48 +1,126 @@
-<tool id="tp_awk_tool" name="Text reformatting" version="@BASE_VERSION@.1">
+<tool id="tp_awk_tool" name="Text reformatting" version="@BASE_VERSION@.2">
     <description>with awk</description>
     <macros>
         <import>macros.xml</import>
     </macros>
     <requirements>
-        <requirement type="package" version="4.2.0">gawk</requirement>
+        <requirement type="package" version="5.0.1">gawk</requirement>
     </requirements>
     <version_command>awk --version | head -n 1</version_command>
     <command>
 <![CDATA[
+        #set $tool_input = 0
         env -i
+        #for env in $envs
+            '$env.name'='$env.val'
+        #end for
         awk
             --sandbox
             -v FS='	'
             -v OFS='	'
             --re-interval
-            -f "$awk_script"
-            "$infile"
-        > "$outfile"
+            -f '$awk_script'
+            #for group in $infiles
+                tool_input=$tool_input
+                #set $tool_input+=1
+                #if $group.input_type_selection.multiple
+                    #set $inputs = $group.input_type_selection.infile
+                    ## Allow capturing the collection id in the program
+                    ## tool_input_id='$inputs.element_identifier'  FIXME collections dont expose their label to tools.
+                #else
+                    #set $inputs = [$group.input_type_selection.infile]
+                #end if
+                #for file in $inputs
+                    tool_input_id='$file.element_identifier'
+                    '$file'
+                #end for
+            #end for
+        > '$outfile'
 ]]>
     </command>
     <configfiles>
         <configfile name="awk_script">$code</configfile>
     </configfiles>
     <inputs>
-        <param name="infile" format="txt" type="data" label="File to process" />
+        <repeat name="infiles" title="Inputs">
```

Comment:
Is a repeat really necessary. You already can process multiple files with `multiple="true"`

Reply:
In this case my preference is always to add a `when` for collection input, because IMHO repeats don't work well in workflows, i.e. only for a fixed number of input collections.

Reply:
Somethin along the following lines: In the input you have 

```
<conditional>
  <param name="intype" ... />
  <when "single"></when>
  <when "single"></when>
  <when "multiple"></when>
  <when "list"></when>
  <when "multiple-list"></when>
</conditional>
````

in the output section you have one output for each of these cases:

```
        <data ...>
            <filter>inselect['intype'] == "single" or inselect['intype'] == "multiple"</filter>
        </data>
        <collection ... format_source="...">
            <discover_datasets pattern="__name__" directory="out" visible="false" />
            <filter>inselect['intype'] == "list" or inselect['intype'] == "list_multiple"</filter>
        </collection>
```

Reply:
That would prevent n:m:n mappings. It would also have the issue of inconsistent output behavior as only the 'single' and 'multiple' options would allow format_source.